spec/rubocop/cop/rails/save_bang_spec.rb

```diff
@@ -0,0 +1,69 @@
+# encoding: utf-8
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe RuboCop::Cop::Rails::SaveBang do
+  subject(:cop) { described_class.new }
+
+  shared_examples 'checks_offense' do |method|
+    it "when using #{method} with arguments" do
+      inspect_source(cop, "object.#{method}(name: 'Tom', age: 20)")
+
+      if method == :destroy
+        expect(cop.messages).to be_empty
+      else
+        expect(cop.messages)
+          .to eq([["Use `#{method}!` instead of `#{method}` ",
+                   'if the return value is not checked.'].join])
```

Comment:
`join` is redundant, btw. You can just have the 2 literals on two adjacent lines and they will automatically be merged into one.